### ISSUE TITLE
No module named 'file_utils'

### ISSUE THREAD

**Issue by seamanj** [CalciferZh/SMPL#27]
when I run SMIL_torch_batch.py, it gave me "No module named 'file_utils" 

Also, find_mini_rgbd and MicroRGBD cannot be found as well. I guess they are from file_utils package.

**Comment by sebftw** [comment]
Yes. It is just some files I used to read the pose data. It was very early in our project, so we have since changed a lot of it. I just wanted to get the code out there.
It works like a torch module, so it is initialized by
`SMILPY = SMIL('./model.pkl', sparse=sparse).to(device)`
And then to get vertices and joint positions:
`v, Jtr = SMILPY(vbeta, vpose, vtrans)`
Where betas, pose and translation values should be given in a batched manner, so e.g. translation dimensions is Nx3, where the first dimension, N, is the batch.

It works with SMPL and its derivatives (SMIL specifically). If you have any trouble feel free to write. :)
Are you working with the SMIL model?

**Comment by seamanj** [comment]
thanks for your reply. 

Yes, I am currently working with SMIL. Could I have the "file_utils" module to get the program run?

Appreciated!

**Comment by sebftw** [comment]
Here are the relevant files we used at the time. [Oldfiles.zip on Drive.](https://drive.google.com/open?id=16s2FcVwFXv47TP5OXD-_cl6GkU8Z40xx)

**Comment by seamanj** [comment]
Much appreciated for sharing! Suggest adding them into the repo so that being more friendly to new users.


BTW, I have an idea and want to ask for your advice. 

I want to register the SMPL to point cloud like this paper "Detailed Full-Body Reconstructions of Moving People from Monocular RGB-D Sequences", Do you think it is possible to finish this job based on your code? Just wondering how to optimise the total energy w.r.t pose and beta parameter.

Many thanks

**Comment by sebftw** [comment]
I am not sure how to register a point cloud to a mesh, but if you make pose and beta into parameters (set requires_grad=True, or wrap in torch.Parameter) the loss
`v, _ = SMILPY(vbeta, vpose, vtrans)`
`loss = energy(v, point_cloud)`
is a function of vbeta and vpose that can be optimized by `loss.backward()` using SGD or another optimization method. I think this is similar to what is done in the paper you refer to.
If you look in the [MINI-RGBD paper](https://www.iosb.fraunhofer.de/servlet/is/82920/Hesse-MINI-RGBD-ECCVW2018.pdf) [sec. 3.1] they also get an energy for the registration (with the method described in the [SMIL paper](https://www.iosb.fraunhofer.de/servlet/is/82920/smil_paper.pdf) [sec. 2]). I would suggest implementing the method described in SMIL/MINI-RGBD, unless there is some feature you need described in "Detailed Full-Body Reconstructions...". I only skimmed through this paper, but it is cited in the both SMIL and MINI-RGBD, so it is likely they used this same method or similar - and expanded upon it.

**Comment by seamanj** [comment]
Thanks a lot. Let me have a try.
